cpp/unreal_plugins/UrdfBot/Source/UrdfBot/UrdfJointComponent.cpp

```diff
@@ -0,0 +1,78 @@
+//
+// Copyright(c) 2022 Intel. Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+//
+
+#include "UrdfJointComponent.h"
+
+#include <iostream>
+
+#include <PhysicsEngine/PhysicsConstraintComponent.h>
+
+#include "CoreUtils/Assert.h"
+#include "UrdfLinkComponent.h"
+#include "UrdfParser.h"
+
+void UUrdfJointComponent::initialize(UrdfJointDesc* joint_desc, UUrdfLinkComponent* parent_link, UUrdfLinkComponent* child_link)
+{
+    this->joint_type_ = joint_desc->type_;
+
+    ConstraintInstance.SetDisableCollision(true);
+
+    ConstraintInstance.ProfileInstance.ConeLimit.Swing1Motion = EAngularConstraintMotion::ACM_Locked;
+    ConstraintInstance.ProfileInstance.ConeLimit.Swing2Motion = EAngularConstraintMotion::ACM_Locked;
+    ConstraintInstance.ProfileInstance.TwistLimit.TwistMotion = EAngularConstraintMotion::ACM_Locked;
+    ConstraintInstance.UpdateAngularLimit();
+
+    ConstraintInstance.ProfileInstance.AngularDrive.AngularDriveMode = EAngularDriveMode::TwistAndSwing;
+    ConstraintInstance.ProfileInstance.ConeLimit.bSoftConstraint = true;
+    ConstraintInstance.ProfileInstance.TwistLimit.bSoftConstraint = false;
+
+    const float M_TO_CM = 100.0f;
+
+    float range = (joint_desc->upper_ - joint_desc->lower_) * 0.5f;
```

Comment:
These variables (`range`, `spring`, etc) should match the names of the parameters to the Unreal functions where they are used. For example, instead of `range`, it should be called `twist_limit_angle` because that is what Unreal calls it. It would be better if these variables were more local to where they are used:

```
    case UrdfJointType::Revolute:
        {
            // DEFINE LOCAL VARS HERE
            // use twist degree for single angular freedom with best PhysX optimization
            ConstraintInstance.SetAngularTwistLimit(EAngularConstraintMotion::ACM_Limited, FMath::RadiansToDegrees(range));
            ConstraintInstance.SetOrientationDriveTwistAndSwing(true, false);
            ConstraintInstance.SetAngularVelocityDriveTwistAndSwing(true, false);
            ConstraintInstance.SetAngularDriveParams(FMath::RadiansToDegrees(spring), FMath::RadiansToDegrees(damping), FMath::RadiansToDegrees(force_limit));
            break;
        }

Reply:
fixed